bag_reader:
  ros__parameters:
    use_sim_time: true
    bag_file_dir: "/root/dddmr_bags/meydan_hotel"
    point_cloud_topic: "/lslidar_point_cloud"
    odometry_topic: "/odom"
    skip_frame: 10

lego_loam_ip:
  ros__parameters:
    use_sim_time: true
    # leishen16
    laser:
        num_vertical_scans: 16
        num_horizontal_scans: 1000
        ground_scan_index: 7
        vertical_angle_bottom: -15.0          # degrees
        vertical_angle_top: 15.0            # degrees
        sensor_mount_angle: 0.0               # degrees
        scan_period: 0.05                   # seconds

    imageProjection:
        segment_valid_point_num: 5
        segment_valid_line_num: 2
        segment_theta: 60.0                      # decrese this value may improve accuracy
        maximum_detection_range: 100.0            # point cloud within this range will be used to project image

lego_loam_fa:
  ros__parameters:
    use_sim_time: true
    laser:
        num_vertical_scans: 16
        num_horizontal_scans: 1000
        scan_period: 0.05                    # seconds

    featureAssociation:
        odom_type: "wheel_odometry"         #wheel_odometry/laser_odometry
        robot_frame: "base_link"
        sensor_frame: "laser_link"
        sensor_x: 0.3
        sensor_y: 0.0
        sensor_z: 0.38
        sensor_roll: 0.0
        sensor_pitch: 0.0 #-0.1745
        sensor_yaw: -3.1415926
        edge_threshold: 0.1
        surf_threshold: 0.1
        nearest_feature_search_distance: 3.0

    mapping:
        to_map_optimization: true

lego_loam_mo:
  ros__parameters:
    use_sim_time: true
    mapping:

        distance_between_key_frame: 1.0
        angle_between_key_frame: 1.0

        enable_loop_closure: true

        surrounding_keyframe_search_num: 20
        history_keyframe_search_radius: 10.0
        history_keyframe_search_num: 5
        history_keyframe_fitness_score: 0.5
        
        ground_voxel_size: 0.3
        ground_edge_threshold_num: 50